### ISSUE TITLE
Schema `__metadata` field validation of `avro.java.string` attribute

### ISSUE THREAD

**Issue by dankraw** [allegro/hermes#1015]
Users tend to add `avro.java.string` attribute to String fields in Avro schemas in order to generate Java classes with proper `String` type instead of Avro's `Utf8`. If the attribute is added to all fields - and by mistake - to `__metadata` field as well (which is a required field), messages produced with this version of schema will not be deserialised by Hermes consumers at all - we get a ` java.lang.ClassCastException: java.lang.String cannot be cast to org.apache.avro.util.Utf8` (even though frontends will accept the incoming messages, wrap them adding custom metadata and publish to Kafka without any problem).

1. We could validate the field in Hermes management if it contains the attribute and disallow such Schema extension for `__metadata` field.
2. The other solution is fix consumers behaviour, and allow both `Map<String, String>` and `Map<Utf8, Utf8>` as `__metadata` target type.

The troublesome field looks as follows:

```js
{
  "name": "__metadata",
  "type": [
    "null",
    {
      "type": "map",
      "values": "string",
      "avro.java.string": "String"
    }
  ],
  "doc": "Field internally used by Hermes to propagate metadata.",
  "default": null
}
```

**Comment by druminski** [comment]
The second one solution is better and it should be easy to implement. As a hint we have integration tests for publishing avro messages.

**Comment by jewertow** [comment]
I will take care of it.

**Comment by jewertow** [comment]
Resolved in #1192
